### PR TITLE
Message interface implementation

### PR DESCRIPTION
**Pull request by sylfabre**
As discussed on #16

### CODE REVIEW

File: MessageTrait.php [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    public function getHeader(string $name, $default = null)
+    {
+        return array_key_exists($name, $this->headers) ? $this->headers[$name] : $default;
+    }
+
+    public function setRedelivered(bool $redelivered): void
+    {
+        $this->redelivered = (bool) $redelivered;
```
Comment (makasim) [inline MessageTrait.php]: `(bool)` is not needed.

File: MessageTrait.php [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    public function setReplyTo(string $replyTo = null): void
+    {
+        $this->setHeader('reply_to', $replyTo);
+    }
+
+    public function getReplyTo(): ?string
+    {
+        return $this->getHeader('reply_to');
+    }
+
```
Comment (makasim) [inline MessageTrait.php]: remove empty line.

File: MessageTrait.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+declare(strict_types=1);
+
+namespace Interop\Queue;
+
+/**
+ * The MessageTrait is a common implementation of the Message interface
+ */
+
+Trait MessageTrait
```
Comment (makasim) [inline MessageTrait.php]: make `Trait` lower case

Comment (makasim) [inline MessageTrait.php]: remove empty line between comment docblock and trait definition

File: MessageTrait.php [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    public function setProperties(array $properties): void
+    {
+        $this->properties = $properties;
+    }
+
+    public function setProperty(string $name, $value): void
+    {
+        $this->properties[$name] = $value;
```
Comment (makasim) [inline MessageTrait.php]: if null value is passed the property (as well as header) should be unset, look at AmqpMessage implementation. https://github.com/queue-interop/amqp-interop/blob/a304a7a876ce872246e8fc01f6dc7b6c86c34277/src/Impl/AmqpMessage.php#L82-L86

**Comment by sylfabre** [comment]
@makasim thank you for the review. I've actually looked at the Redis implementation as source about both your unnecessary `(bool)` and `null` value for property comments. A single Trait will clean all these differences :)

File: src/MessageTrait.php [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    public function setProperties(array $properties): void
+    {
+        $this->properties = $properties;
+    }
+
+    public function setProperty(string $name, $value): void
+    {
+        if(null === $value){
```
Comment (makasim) [inline src/MessageTrait.php]: should be `) {`

File: src/MessageTrait.php [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+            unset($this->properties[$name]);
+        } else{
```
Comment (makasim) [inline src/MessageTrait.php]: should be `else {`

**Comment by sylfabre** [comment]
@makasim fixed !

**Pull request merged**